src/server_manager/web_app/app.ts

```diff
@@ -920,24 +956,107 @@ export class App {
         });
   }
 
+  private async setAccessKeyDataLimit(limit: server.DataLimit) {
+    if (!limit) {
+      return;
+    }
+    const previousLimit = this.selectedServer.getAccessKeyDataLimit();
+    if (previousLimit && limit.bytes === previousLimit.bytes) {
+      return;
+    }
+    const serverView = this.appRoot.getServerView(this.appRoot.selectedServer.id);
+    try {
+      await this.selectedServer.setAccessKeyDataLimit(limit);
+      this.appRoot.showNotification(this.appRoot.localize('saved'));
+      serverView.accessKeyDataLimit = this.dataLimitToDisplayDataAmount(limit);
+      this.refreshTransferStats(this.selectedServer, serverView);
+    } catch (error) {
+      console.error(`Failed to set access key data limit: ${error}`);
+      this.appRoot.showError(this.appRoot.localize('error-set-data-limit'));
+      serverView.accessKeyDataLimit = this.dataLimitToDisplayDataAmount(
+          previousLimit || await this.computeDefaultAccessKeyDataLimit(this.selectedServer));
```

Comment:
IMO it's cleaner if we call `this.computeDefaultAccessKeyDataLimit(this.selectedServer, await this.selectedServer.listAccessKeys())` here and make the second parameter required.  Or alternatively get rid of the second parameter completely and call `listAccessKeys()` twice above